src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -313,6 +313,15 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                 case BoundKind.IndexerAccess:
                     return CheckPropertyValueKind(node, expr, valueKind, checkingReceiver, diagnostics);
 
+                case BoundKind.IndexOrRangePatternIndexerAccess:
+                    if (((BoundIndexOrRangePatternIndexerAccess)expr).PatternSymbol.Kind == SymbolKind.Property)
+                    {
+                        // If this is an Index indexer PatternSymbol should be a property, pointing to the
+                        // pattern indexer. If it's a Range access, it will be a method, pointing to a Slice method
```

Comment:
>If it's a Range access, it will be a method, pointing to a Slice method [](start = 44, length = 71)

NIT: This comment is a little confusing (at least for me at first read). Could you clarify that we'll instead handle range below?